CaloMC/src/CaloDigiMaker_module.cc

```diff
@@ -274,7 +273,7 @@ namespace mu2e {
 	    int sampleStop  = hitStops[ihit];
```

Comment:
The logic here is difficult to follow, due in part to moving between iterators and indices.  Why not stick with one or the other?  I also see no point in moving the amplitude test outside the 'while' condition.  I don't understand why some indices are typed 'unsigned', some 'int'.  They should all be size_t.

Reply:
Yeah, yeah, I'll fix this...

Reply:
The FastCAloRecoDigi was written completely independently from the rest of the reco with different assumptions about the hit timing. I'll rewrite it to make it consistent first, and then we can decide how to synchronize with the tracker/CRV (PBT and others)

Reply:
I thought that using that module would have made life simpler as a temporary solution, BUT if you want to rewrite it, I think the best thing to do is to fix the [`raw_processor`](https://github.com/Mu2e/Offline/blob/master/CaloReco/inc/CaloRawWFProcessor.hh) class and configure the standard [CaloRecoDigi_module](https://github.com/Mu2e/Offline/blob/master/CaloReco/src/CaloRecoDigiMaker_module.cc) to use it. This would probably allow less maintenance in the long term.

Reply:
The module is this [one](https://github.com/Mu2e/Offline/blob/master/DAQ/src/CaloHitsFromFragments_module.cc), BUT for fixing the raw_processor class I don't think you need it.
The raw_processor needs to use the the peak present in the digi to evaluate the time and the energy. Right know, it loops over the entire waveform because at the time when it was written the CaloDifi didn't have the peak info.